### PR TITLE
Preserve object order when swapping #1302

### PR DESCRIPTION
**Pull request by IanTeo**
Fixes: #1302 

Original Shapes - numbers represent their ordering:
image

Old behavior - Shapes are swapped, but their ordering is not swapped:
image

New behavior - Shapes AND ordering are swapped:
image

### CODE REVIEW

**Comment by IanTeo** [comment]
Tests pass on 2013 (Except the flaky tests waiting to be merged)

**Comment by IanTeo** [comment]
Change UpdateZOrder to SwapZOrder, because UpdateZOrder can cause the z-order of other objects on the slide to become messed up.

**Comment by initialshl** [comment]
Works on PPT 2016.
Positions Lab tests have to be updated to check for z-order too.

**Comment by initialshl** [comment]
All tests passed on PPT 2016

File: PowerPointLabs/PowerPointLabs/PositionsLab/PositionsPaneWPF.xaml.cs [hunk 1]
```diff
@@ -1666,6 +1674,39 @@ private void SaveSelectedShapePositions(PowerPoint.ShapeRange shapes, Dictionary
                 dictionary.Add(s.Id, new PositionShapeProperties(new System.Drawing.PointF(s.Left, s.Top), s.Rotation, s.HorizontalFlip, s.VerticalFlip));
             }
         }
+
+        private void UpdateZOrder(Shape original, int zOrder)
```
Comment (kai33) [inline PowerPointLabs/PowerPointLabs/PositionsLab/PositionsPaneWPF.xaml.cs]: this ZOrder methods look generic, and can be used by other features.
try to put it into the util class

**Comment by IanTeo** [comment]
I wrapped the shapes with PPShape and used the PPShape methods instead of duplicating the codes

**Comment by IanTeo** [comment]
Erm.. I am not sure why the test is failing @Skaty I don't understand the error message. What should I be doing?

**Comment by IanTeo** [comment]
This PR was created from the wrong branch. Recreating another PR

**Pull request closed without merging**